SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp

```diff
@@ -503,6 +505,23 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
 		return FReply::Handled();
 	}
 
+	if (!bHasAttemptedAuth)
+	{
+		FString SpatialInfoArgs = TEXT("auth login");
+		FString SpatialInfoResult;
+		FString StdErr;
+		int32 ExitCode;
+		FPlatformProcess::ExecProcess(*SpatialGDKServicesConstants::SpatialExe, *SpatialInfoArgs, &ExitCode, &SpatialInfoResult, &StdErr);
```

Comment:
Will this block the entire editor until the user logs in? Can you test this by logging out of your google account and then giving it a run please <3

Reply:
I assume you mean logging out of spatial with "spatial auth logout"?

If you did... Perhaps it locks the editor but I could not tell. If it does the process is so short I did not have time to notice the editor locked and I was trying.

Reply:
Could you move this code to some `Authorise` function?

Reply:
I was thinking about it to somewhere globally accessible. I guess would have if we used it somewhere else. I didn't see the need in terms of readability but if you feel it would be more readable as a function I can add it.

Reply:
I think it should be nicer if it's elsewhere and we only have one line to call `Authorise` function here. If you prefer to keep it as is - I wouldn't block the PR on that.

Reply:
No. I mean literally log out of your google account. This is how `spatial auth login` actually generates the token, it will use your SAML from Google to generated an auth token. If you're not logged into google it will present you with a login page. I'm asking if this login page blocks the Engine.

Reply:
The engine is blocked but I never actually got a popup to log into google. I had to do it through the browser. What do?

Reply:
The correct approach would be to run the login in a separate thread and have the processes which rely on the login wait for a delegate which gets called when login is complete. Perhaps we can check for token existence in the app data folder?